### PR TITLE
feat(pie): Added showPercentageDecimals to Pie

### LINKED ISSUE #3077: Pie Chart Decimal Places
Can we have the ability to specify the number of decimal places when using pie charts and "showSeriesPercentage(true)"?  Our requirements specify for us to show 2 decimal places.

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes #3077

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>
## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [ ] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

## Testing:
I tested the new publish parameter in the dermatology page.

### CODE REVIEW

File: packages/chart/src/Pie.ts [hunk 1]
```diff
@@ -411,3 +415,4 @@ Pie.prototype.publish("innerRadius", 0, "number", "Sets inner pie hole radius as
 Pie.prototype.publish("minOuterRadius", 20, "number", "Minimum outer radius (pixels)");
 Pie.prototype.publish("startAngle", 0, "number", "Starting angle of the first (and largest) wedge (degrees)");
 Pie.prototype.publish("labelHeight", 12, "number", "Font size of labels (pixels)");
+Pie.prototype.publish("percentageDecimals", 1, "number", "Number of decimal places to display while showing series percentage", null, {disable: w => !w.showSeriesPercentage()});
```
Comment (GordonSmith) [inline packages/chart/src/Pie.ts]: Can you change this so it matches what we do with the tooltip values:  https://github.com/hpcc-systems/Visualization/blob/master/packages/api/src/ITooltip.ts#L154

Comment (jbrundage) [reply packages/chart/src/Pie.ts]: Hmm, yeah that makes sense... I'm on it

**Comment by GordonSmith** [comment]
@jbrundage please view all related gallery items...

File: packages/chart/src/Pie.ts [hunk 2]
```diff
@@ -123,6 +125,7 @@ export class Pie extends SVGWidget {
         const context = this;
         this.updateD3Pie();
         this._palette = this._palette.switch(this.paletteID());
+        this._percentageFormatter = d3Format(this.percentageFormat() as string);
```
Comment (GordonSmith) [inline packages/chart/src/Pie.ts]: `as string` should not be needed?

File: packages/chart/src/Pie.ts [hunk 3]
```diff
@@ -403,8 +406,12 @@ export interface Pie {
     startAngle(_: number): this;
     labelHeight(): number;
     labelHeight(_: number): this;
+    percentageFormat(): string;
+    percentageFormat(_: string): this;
+
```
Comment (GordonSmith) [inline packages/chart/src/Pie.ts]: remove blank line

File: packages/chart/src/Pie.ts [hunk 4]
```diff
@@ -403,8 +406,12 @@ export interface Pie {
...
 }
 Pie.prototype.publish("showSeriesPercentage", false, "boolean", "Append data series percentage next to label");
+Pie.prototype.publish("percentageFormat", ",.0f", "string", "Number format used for formatting values", null, {disable: w => !w.showSeriesPercentage()});
```
Comment (GordonSmith) [inline packages/chart/src/Pie.ts]: Rename to "seriesPercentageFormat" and change description to mention "series" as well.

**Comment by jbrundage** [comment]
@GordonSmith this PR is revised and ready for a review

**Pull request merged**